File: MATTHEW/leetcode/AIcoding/1.Gemini-2.5/cpp_solutions/3432-count-partitions-with-even-sum-difference.cpp
#include <vector>
#include <numeric>

class Solution {
public:
    int countPartitions(std::vector<int>& nums) {
        long long total_sum = 0;
        for (int num : nums) {
            total_sum += num;
        }

        if (total_sum % 2 != 0) {
            return 0;
        }

        long long n = nums.size();
        long long MOD = 1e9 + 7;

        long long result = 1;
        long long base = 2;
        long long power = n;

        while (power > 0) {
            if (power % 2 == 1) {
                result = (result * base) % MOD;
            }
            base = (base * base) % MOD;
            power /= 2;
        }

        return static_cast<int>(result);
    }
};

File: leetcode/AIcoding/3.DeepSeek-v3/java_solutions/3432-count-partitions-with-even-sum-difference.java
class Solution {
    public int countPartitions(int[] nums) {
        int MOD = 1_000_000_007;
        int totalSum = 0;
        for (int num : nums) {
            totalSum += num;
        }
        if (totalSum % 2 != 0) {
            return 0;
        }
        int target = totalSum / 2;
        int n = nums.length;
        int[][] dp = new int[n + 1][target + 1];
        dp[0][0] = 1;
        for (int i = 1; i <= n; i++) {
            for (int j = 0; j <= target; j++) {
                dp[i][j] = dp[i - 1][j];
                if (j >= nums[i - 1]) {
                    dp[i][j] = (dp[i][j] + dp[i - 1][j - nums[i - 1]]) % MOD;
                }
            }
        }
        return dp[n][target];
    }
}